File: day01/1.py
#!/bin/python3

file = open("input.txt")
sum = 0
for line in file:
    for c in line:
        if c.isdigit():
            sum += int(c)*10
            break;
    for c in reversed(line):
        if c.isdigit():
            sum += int(c)
            break;
print(sum)

File: day01/1.cpp
#include <algorithm>
#include <cctype>
#include <cstdint>
#include <fstream>
#include <iostream>

using namespace std;

int main(int , char **) {
    fstream f{"input.txt"};
    string line;
    uint64_t sum = 0;
    while (getline(f, line)) {
        auto f = find_if(begin(line), end(line), ::isdigit);
        auto l = find_if(rbegin(line), rend(line), ::isdigit);
        sum += (*f - '0') * 10 + (*l - '0');
    }
    std::cout << sum << std::endl;
    return 0;
}
